addons/mail/static/src/models/message/message.js

```diff
@@ -80,6 +73,21 @@ function factory(dependencies) {
             }
 
             // relations
+            // This needs to be the first to edit data2.threadCaches as it's
+            // data coming from server and any client-side modification should
+            // be done after it.
```

Comment:
add something like "because the 'replace' command would override any preceding change"